### ISSUE TITLE
Enter key not detected in key event

### ISSUE THREAD

**Issue by tomm13** [lawsie/guizero#455]
As the title suggests, when I assigned a function to the when_key_pressed property of a textbox, the enter key is detected as space.

Steps to reproduce the behaviour:
```
def enterKeyClicked(event):
    print("key:" +  str(event.key))

app = App()

textbox = TextBox(app)
textbox.when_key_pressed = enterKeyClicked

app.display()
```
This outputs the correct symbol for most of keys but keys such as tab enter or delete, will return " "

Expected behavior: returns "enter" and "tab" respectively when pressed instead of " ".

 - OS: macOS 12.6
 - guizero version: 1.3.0

**Comment by martinohanlon** [comment]
guizero simply returns what was is given to it by tkinter (the underlying GUI framework). If tkinter doesnt return these characters then there probably isnt a lot which can be done.

https://github.com/lawsie/guizero/blob/master/guizero/event.py#L33

The event object return also includes the underlying tk_event object which you may be able to interrogate to get at the non-printing characters. e.g.

```python
 def enterKeyClicked(event):
    print("tk event object", event.tk_event)

app = App()

textbox = TextBox(app)
textbox.when_key_pressed = enterKeyClicked

app.display()
```

**Comment by martinohanlon** [comment]
You can get the keycode from the `tk_event` which includes non-printing charactors.

```python
from guizero import App, TextBox

def enterKeyClicked(event):
    print("keycode:" + str(event.tk_event.keycode))

app = App()

textbox = TextBox(app)
textbox.when_key_pressed = enterKeyClicked

app.display()
```

Suggest we add `tk_event.keycode` to the guizero event object at some point. Its a minor change to the `event.EventData` object.

**Comment by lawsie** [comment]
Now in dev

**Comment by martinohanlon** [comment]
Released in v1.4.0
